### DISCUSSION TITLE
Passing typescript interfaces as the value of a props to the component.

### DISCUSSION THREAD

**Discussion by mrc-bsllt** [vuejs/rfcs#606] [General]
It would be nice to be able to pass the interface, type etc. declared with typescript as the value of a props:

```javascript
<script setup lang="ts">
interface MyInterface {
 name: string;
}
</script>

<template>
 <div>
  <MyCustomCompent :generic-type="MyInterface" />
 </div>
</template>
```

**Comment by paulmelero** [comment]
The problem is that interfaces/types are information that disappears in runtime. What would be the value of `MyInterface.name` in runtime?

One thing is defining prop types and consuming prop values is a different thing.
